hdf5-sys/build.rs

```diff
@@ -297,35 +300,55 @@ mod macos {
         }
         // We have to explicitly support homebrew since the HDF5 bottle isn't
         // packaged with pkg-config metadata.
-        let (v18, v110) = if let Some(version) = config.version {
-            (version.major == 1 && version.minor == 8, version.major == 1 && version.minor == 10)
+        let (v18, v110, v112) = if let Some(version) = config.version {
+            (
+                version.major == 1 && version.minor == 8,
+                version.major == 1 && version.minor == 10,
+                version.major == 1 && version.minor == 12,
+            )
         } else {
-            (false, false)
+            (false, false, false)
         };
         println!(
             "Attempting to find HDF5 via Homebrew ({})...",
             if v18 {
                 "1.8.*"
             } else if v110 {
                 "1.10.*"
+            } else if v112 {
+                "1.12.*"
             } else {
                 "any version"
             }
         );
-        if !v18 {
+        if !(v18 || v110) {
```

Comment:
There's tons of copy/paste now and weird logic in the whole homebrew section of this, should probably be rewritten so it's cleaner... (also `hdf5-mpi` is only available for 1.12 and later).

(could be done separately outside of this PR, just noting)

Reply:
#112